Developer docs update

mlacko64 opened the pull request:
Added note that RKE2/Rancher have to be fully up before continuing with next steps. In case commands are executed too fast (before RKE2/Rancher are fully up) Harvester will not be accessible.

FrankYang0529 commented on docs/developer/developer-mode-installation.md:
```diff
@@ -58,16 +58,19 @@ You can install Harvester on an [RKE2](https://docs.rke2.io/) cluster using the
+:::
+
+:::info note
+If you are unable to locate the `kubectl` binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.
```
May you put two note in one `info note` block? Original render result is like following:

![Screenshot 2024-08-21 at 1 36 46 PM](https://github.com/user-attachments/assets/2c14aa1f-e1a6-49c7-8265-57e7df7cbd8a)

jillian-maroket replied:
@FrankYang0529 Thanks for reviewing. I'll clean up the wording and markup when I do my own review. Which doc versions should be updated?

FrankYang0529 replied:
I think we can also apply this to v1.3 document. Thanks.

FrankYang0529 reviewed (approved):
Overall LGTM. Leave a minor comment.

jillian-maroket commented on docs/developer/developer-mode-installation.md:
````diff
@@ -58,16 +58,19 @@ You can install Harvester on an [RKE2](https://docs.rke2.io/) cluster using the
     sudo chown $(id -u):$(id -g) ~/.kube/config
     ```
+:::info note
+Wait until RKE2 will be fully up (all pods in `kube-system` namespace will be running) before continuing with next steps.
+:::
+
+:::info note
+If you are unable to locate the `kubectl` binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.
````
```suggestion
:::info important

Allow some time for all pods in the `kube-system` namespace to reach the `Ready` state before proceeding to the next step.

If you are unable to locate the kubectl binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.

```

jillian-maroket reviewed (commented):
Please use the suggested wording and markup (only one note for both sentences), and apply the same changes to the v1.3.0 document. Thank you for contributing.

mlacko64 commented:
Hello,
thanks for reviewing; I have used suggested wording and markup and also added same change to 1.3 docs file. Squashed commits to one and updated commit message.

jillian-maroket commented on docs/developer/developer-mode-installation.md:
````diff
@@ -58,16 +58,20 @@ You can install Harvester on an [RKE2](https://docs.rke2.io/) cluster using the
     sudo chown $(id -u):$(id -g) ~/.kube/config
     ```
+    
+:::info important
+
+Allow some time for all pods in the `kube-system` namespace to reach the `Ready` state before proceeding to the next step.
+
+If you are unable to locate the kubectl binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.
+
+:::
````
```suggestion
    :::info important

    Allow some time for all pods in the `kube-system` namespace to reach the `Ready` state before proceeding to the next step.

    If you are unable to locate the kubectl binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.

    :::
```

jillian-maroket replied:
The current indentation is causing formatting issues.

![Screenshot 2024-08-22 at 10 30 06 PM](https://github.com/user-attachments/assets/e60e585a-8286-4733-82c7-2fea09028aec)

jillian-maroket commented on docs/developer/developer-mode-installation.md:
````diff
@@ -92,6 +96,10 @@ If you are unable to locate the `kubectl` binary in `/usr/local/bin`, check `/va
     --set bootstrapPassword=admin
     ```
 
+:::info note
+Wait until Rancher will be fully up (all pods in `cattle-system` namespace will be running) before continuing with next steps.
+:::
````
```suggestion
    :::info important

    Allow some time for all pods in the `cattle-system` namespace to reach the `Ready` state before proceeding to the next step.

    :::
```

jillian-maroket commented on versioned_docs/version-v1.3/developer/developer-mode-installation.md:
````diff
@@ -59,15 +59,19 @@ You can install Harvester on an [RKE2](https://docs.rke2.io/) cluster using the
     sudo chown $(id -u):$(id -g) ~/.kube/config
     ```
 
+:::info important
+
+Allow some time for all pods in the `kube-system` namespace to reach the `Ready` state before proceeding to the next step.
+
+If you are unable to locate the kubectl binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.
+
+:::
````
```suggestion
    :::info important

    Allow some time for all pods in the `kube-system` namespace to reach the `Ready` state before proceeding to the next step.

    If you are unable to locate the kubectl binary in `/usr/local/bin`, check `/var/lib/rancher/rke2/bin`.

    :::
```

jillian-maroket commented on versioned_docs/version-v1.3/developer/developer-mode-installation.md:
````diff
@@ -92,6 +96,10 @@ If you are unable to locate the `kubectl` binary in `/usr/local/bin`, check `/va
     --set bootstrapPassword=admin
     ```
 
+:::info note
+Wait until Rancher will be fully up (all pods in `cattle-system` namespace will be running) before continuing with next steps.
+:::
````
```suggestion
    :::info important

    Allow some time for all pods in the `cattle-system` namespace to reach the `Ready` state before proceeding to the next step.

    :::
```

jillian-maroket reviewed (commented):
I fixed the indentation issue and reviewed the new note.

jillian-maroket reviewed (approved):
LGTM. Thanks again for contributing.
